src/Symfony/Component/Console/Helper/TreeHelper.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * The TreeHelper class provides methods to display tree-like structures.
+ *
+ * @author Simon André <smn.andre@gmail.com>
+ *
+ * @implements \RecursiveIterator<int, TreeNode>
+ */
+final class TreeHelper implements \RecursiveIterator
+{
+    private readonly TreeStyle $style;
+
+    private readonly TreeNode $node;
+
+    private \Iterator $childrenIterator;
+
+    public function __construct(
+        private readonly OutputInterface $output,
+        ?TreeNode $node = null,
+        ?TreeStyle $style = null,
+    ) {
+        $this->node = $node ?? new TreeNode();
+        $this->style = $style ?? TreeStyle::default();
+        $this->childrenIterator = new \IteratorIterator($this->node->getChildren());
+        $this->childrenIterator->rewind();
+    }
+
+    public static function create(OutputInterface $output, iterable $values, ?string $root = null, ?TreeStyle $style = null): self
+    {
+        return new self($output, TreeNode::fromValues($values, new TreeNode($root ?? '')), $style);
+    }
+
+    /**
+     * @internal
```

Comment:
I don't think methods should be internal. Why should they be?

Reply:
😶‍🌫️ ux habit  

(removed)